### ISSUE TITLE
[Accessibility] Add "document" role to webview widget

### ISSUE THREAD

**Issue by jooyoungseo** [microsoft/vscode#132275]
CC @isidorn 

Currently, the `webview` widget in VSCode does not have "document" role, causing screen-reader users to manually turn on browse mode in order to read web content when populated.

Please add "document" role like below:

```html
<!-- In the webview div -->
<div role = "document">
    ...
</div>
```

This will improve both user experience and accessibility with screen readers in the web view.

**Comment by isidorn** [comment]
@jooyoungseo thanks a lot for opening this issue. This makes great sense to me.
I think webviews are mostly read only, so having a document role be the default is the way to go here.

@mjbvz what do you think about this change? Is it ok if we try this out in September?

**Comment by mjbvz** [comment]
I'm not familiar with `role="document". Does using it make sense for all webviews, including  those that are more application like?

Also where should this be added? Inside the webview itself or on the element that contains the webview?

**Comment by isidorn** [comment]
@mjbvz the `role="document" makes sense for read-only content since the screen reader would switch to a `browse` mode which makes it easier for users to read and browse.
My intuition is telling me that most webviews are readonly and thus we can add this role by default. Do we know if a webview is readonly or not? Do we have some data for this maybe?

I do not think it will make a difference if we add a role on the webview or on the parent element. Whatever you think makes more sense since the screen reader should walk up the parent chain. I feel like it should be on the webview (if vscode controls that html element).

**Comment by mjbvz** [comment]
I will make `role="document"` the default and let extensions override it if needed. Will check in a fix shortly but I will need help verifying it works as expected

**Comment by isidorn** [comment]
Thanks a lot 👏  @mjbvz 
@jooyoungseo can you please give this a try with VS Code insiders from tomorrow (Thursday) and let us know if it behaves as you would expect? We really appreciate your help

**Comment by jooyoungseo** [comment]
Thank you very much for the prompt improvement! It works nicely with NVDA! I have tested in the following environment:

Version: 1.61.0-insider (system setup)
Commit: 34861b8c8aa76b517f203743ad5b5d72ef5fcd81
Date: 2021-09-08T05:17:18.073Z
Electron: 13.1.8
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Windows_NT x64 10.0.19043

**Comment by isidorn** [comment]
@jooyoungseo lovely, thanks for trying it out -> verified.
